### DISCUSSION TITLE
Staging Significantly Faster than Production

### DISCUSSION THREAD

**Discussion by KevinCase** [supabase/supabase#15700] [Questions]
I have two projects in Supabase, one for staging and one for production. I noticed that the staging site loads pages significantly faster than our production site. I am seeing a big discrepancy in time to first byte between the two projects. I'm getting ~400 ms for a simple select query on production and ~100 ms on staging. Our staging server is set to AWS East and our production server is set to AWS West. I'm located in the East Coast. 

Could the difference in server location be causing that big of a discrepancy? Or could there be some configuration differences causing this lag? Our production project is a Pro tier plan and our staging project is a Free tier. We also recently (yesterday) pushed quite a few changes to production that have been on staging for a while. Do the Postgres optimizations take this long to come into effect? Any help is appreciated.

**Comment by GaryAustin1** [comment]
As to the first part you should be able to get a rough idea of latency to AWS with something like this https://clients.amazonworkspaces.com/Health.html although I've seen other tools.   If these are client to REST API requests they are also usually two roundtrips each because of CORS.  So double that latency.   I would expect 100 to 200 msec of latency for the first real data difference is possible, depending of course on how far you are from the AWS sites.  

In order to investigate other differences you would need to run some explain operations on the two servers to see if your query times are different.  You might even be able to look in the Query performance report and calculate the time for some common queries, if you can identify them.
